### DISCUSSION TITLE
Display the name of the failed download

### DISCUSSION THREAD

**Discussion by p858snake** [aclap-dev/video-downloadhelper#948] [Help and Q&A]
### Your question

In the old UI the failed download would show in the error tab so you could work out what it was. 

The new UI only shows the error message

![Capture2](https://github.com/aclap-dev/video-downloadhelper/assets/1076179/a9115003-d459-4236-b9b6-819972031e58)

### Addon & CoApp version and details

version: 9.0.2.12
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101 Firefox/127.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\XXXXXXXXX\\Downloads"
        }
    },
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 570,
    "successfull_dl": 43,
    "never_show_successfull_dl_message": true
}

**Comment by ThomasBecker42** [comment]
I have the same problem. When downloading multiple videos and an "Download failed" I have no chance to see which one it was. 
With the old UI/Version I was able to see the name of the video. 
Firefox 127.0.2 (64-Bit)
Version 9.0.2.12 (stable), target mozilla, locale de.

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

The following is advice based on my experience.  This is the way I've always used VDH going back several years, meaning in releases before the current VDH 9, but also including VDH 9.

It is possible to launch a download off a web page & then close the page.  Don't close the page.  Leave the page open until you've had a chance to verify that the download off that page has completed successfully.  If it has not downloaded successfully, you've still got the web page open.  Reload that page & do the download again.

Always use `Download As`, never `Download`.  `Download As` offers you the possibility of changing the file name of the target file VDH will download into.  Use that feature.  Name your target files with something meaningful.  Don't always blindly accept the name VDH proposes.  Quite often those names are not descriptive, especially if you're downloading multiple videos from the same web page.

Now, I agree that the error message should at least display something to allow you to differentiate which download has failed.  But if there's multiple downloads off the same page, even that will probably not be good enough.  VDH tends to give the same name to every download off a particular web page.  You should still follow my advice above & always manually verify each download.  Don't accept any messages from VDH or the operating system that things are fine.  Load the video up into your video player & skim it to make sure it really did download completely.  Only then can you remove its entry from the VDH menu & close the web page it came from.

Beyond that, I don't like this new error reporting message that doesn't tell you anything.  Fine, sure, I can report the error.  It goes off to some unknown location to be reviewed on some schedule we have no knowledge of.  What exactly gets reported?  At the moment, this is secret.  It should not be secret.  We should see the error message ourselves, like we always have in past releases.  I understand that the idea of the `Report` button is to decrease the volume of problem reports in this forum.  But if we could see our own error messages, we might be able to figure out what went wrong.  It might actually be a user error that we could correct on our own without reporting the problem.  We should see the information we will be reporting if we click the `Report` button.  There are paranoid people who will refrain from clicking the `Report` button precisely because what it being reported is being kept secret.  I can't blame such people, although I'm not quite that paranoid.  But I am one of those people who would rather try to solve my own problems instead of just foisting them on somebody else.  So we should regain a proper error message, even if in the end we just click the `Report` button.

**Reply by schmidtp72** [reply]
I know you're not the developer, but couldn't this be a feature update that when the download fails while using the companion app that it displays at least the selected named output file? For this very reason I try and not use the companion app while downloading in favour of just the browser download. At least that way I can restart the download and if it still won't download (due to a page timeout), I can pull that page back up and add a new download entry. Unfortunately thats what tends to happen with the companion app if there's too many queued downloads the link might expire and you need to refresh. But without the file name is near impossible.

**Reply by GrampaWildWilly** [reply]
I hear you.  I said it myself, that the error message is singularly uninformative.  But there are too many reasons to just keep the page you're downloading from open.  Manually name each download file yourself, so you can tell them all apart.  Manually verify each download yourself.  Manually.  I am not sure there's any way to automate this verification process.  It's really up to you to verify that you downloaded what you expected.

I have hit the timeout issue myself.  There's no way around that.  If your Internet connection is slow, this is just a fact of life.  The only good defense against that is the approach I outlined in my previous post.  But a slow Internet connection is not the only reason pages might time out.  If the serving site is throttling your service, you get the same effect.  So it's really up to you to try to immunize yourself against such events.  Do things the way I advise.  It has worked for me for years.  There's really no way around it.

I will say that when the VDH menu gets one of those green buttons saying a download has completed, at least the item's name is now the file name on your system & not whatever name was detected off the web page.  This is an improvement.  But I think it's only partial.  That name should include the directory path on your system as well as the file name.  Files can have the same name in different directories.  The only way to differentiate them is through their full file specification, which is to say the full directory path.

Still, I will agree with you that the error message needs to be (a) more informative, (b) visible to users, not just hidden behind a `Report` button.

**Reply by schmidtp72** [reply]
I started using JDownloader instead for bulk downloads instead of queuing everything through VDH. Unfortunately with VDH, you don't know whether the download fully completes while using the native app (as sometimes the result says complete but only a fraction of the file was actually downloaded). Only when downloading using the browser download instead can you see error (which I prefer if I can). At least I can see if it stops and I can restart. With JDownloader, if the link goes dry the original site file name is still there and you can go back to refresh through your browser history.
